pyanaconda/ui/gui/gk_keyboard_manager.py

```diff
@@ -0,0 +1,136 @@
+#
+# Copyright (C) 2024 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+from pyanaconda.core.signal import Signal
+from pyanaconda.keyboard import join_layout_variant, parse_layout_variant, KeyboardConfigError
+from pyanaconda.modules.common.constants.services import GK_INPUT_SOURCES
+
+
+class GkKeyboardManager(object):
+    """Class wrapping GNOME Kiosk's input sources API."""
+
+    def __init__(self):
+        self.compositor_selected_layout_changed = Signal()
+        self.compositor_layouts_changed = Signal()
+
+        object_path = GK_INPUT_SOURCES.object_path + '/InputSources/Manager'
+        self._proxy = GK_INPUT_SOURCES.get_proxy(object_path=object_path)
+        self._proxy.PropertiesChanged.connect(self._on_properties_changed)
+
+    def _on_properties_changed(self, interface, changed_props, invalid_props):
+        for prop in changed_props:
+            if prop == 'SelectedInputSource':
+                self.compositor_selected_layout_changed.emit()
+            if prop == 'InputSources':
+                self.compositor_layouts_changed.emit()
+
+    def _path_to_layout(self, layout_path):
+        """
+        Transforms a layout path as returned by GNOME Kiosk to "layout (variant)".
```

Comment:
We don't use this documentation standard.

Please leave the summary (first line) together with opening `"""` as it is here:
https://github.com/rhinstaller/anaconda/blob/master/pyanaconda/modules/storage/installation.py#L169